lib/std/json.zig

```diff
@@ -1626,8 +1626,8 @@ fn parseInternal(
                     var r: T = undefined;
                     const source_slice = stringToken.slice(tokens.slice, tokens.i - 1);
                     switch (stringToken.escapes) {
-                        .None => mem.copy(u8, &r, source_slice),
-                        .Some => try unescapeValidString(&r, source_slice),
+                        .None => if (r.len == source_slice.len) mem.copy(u8, &r, source_slice) else return error.LengthMismatch,
+                        .Some => if (r.len == stringToken.decodedLength()) try unescapeValidString(&r, source_slice) else return error.LengthMismatch,
```

Comment:
Small nitpick but this would be easier to read if you moved the check on `decodedLenght` before the switch since it works for both cases.

Reply:
made chang here https://github.com/ziglang/zig/pull/14370/commits/75b0db9ce806fc69be6e829036eac66a79ea54ee